Index out of bound exception in tracking database

https://acra.mytfg.de/acralyzer/_design/acralyzer/index.html#/report-details/ibis/8f21fcbb-24b7-424d-9e3c-74725caa8cf8
```
java.lang.RuntimeException: Unable to stop service de.mytfg.jufo.ibis.Tracking@39dd5dc0: android.database.CursorIndexOutOfBoundsException: Index 1 requested, with a size of 1
at android.app.ActivityThread.handleStopService(ActivityThread.java:2943)
at android.app.ActivityThread.access$2200(ActivityThread.java:148)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1395)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:135)
at android.app.ActivityThread.main(ActivityThread.java:5312)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:901)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:696)
Caused by: android.database.CursorIndexOutOfBoundsException: Index 1 requested, with a size of 1
at android.database.AbstractCursor.checkPosition(AbstractCursor.java:426)
at android.database.AbstractWindowedCursor.checkPosition(AbstractWindowedCursor.java:136)
at android.database.AbstractWindowedCursor.getDouble(AbstractWindowedCursor.java:86)
at de.mytfg.jufo.ibis.GPSDatabase.sendToServer(GPSDatabase.java:109)
at de.mytfg.jufo.ibis.Tracking.stopOnlineTracking(Tracking.java:218)
at de.mytfg.jufo.ibis.Tracking.disconnect(Tracking.java:249)
at de.mytfg.jufo.ibis.Tracking.onDestroy(Tracking.java:243)
at android.app.ActivityThread.handleStopService(ActivityThread.java:2926)
... 9 more
android.database.CursorIndexOutOfBoundsException: Index 1 requested, with a size of 1
at android.database.AbstractCursor.checkPosition(AbstractCursor.java:426)
at android.database.AbstractWindowedCursor.checkPosition(AbstractWindowedCursor.java:136)
at android.database.AbstractWindowedCursor.getDouble(AbstractWindowedCursor.java:86)
at de.mytfg.jufo.ibis.GPSDatabase.sendToServer(GPSDatabase.java:109)
at de.mytfg.jufo.ibis.Tracking.stopOnlineTracking(Tracking.java:218)
at de.mytfg.jufo.ibis.Tracking.disconnect(Tracking.java:249)
at de.mytfg.jufo.ibis.Tracking.onDestroy(Tracking.java:243)
at android.app.ActivityThread.handleStopService(ActivityThread.java:2926)
at android.app.ActivityThread.access$2200(ActivityThread.java:148)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1395)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:135)
at android.app.ActivityThread.main(ActivityThread.java:5312)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:901)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:696)
```

--- a/app/src/main/java/de/mytfg/jufo/ibis/GPSDatabase.java
+++ b/app/src/main/java/de/mytfg/jufo/ibis/GPSDatabase.java
@@ -102,31 +102,32 @@ public class GPSDatabase {
         double cutDistEnd = random.nextInt(80) + 20d;
         double distBegin = 0d;
         double distEnd = 0d;
-        int cutColIdBegin;
-        int cutColIdEnd;
+        int cutColIdBegin = -1;
+        int cutColIdEnd = -1;
         cursor.moveToFirst();
-        while(true) {
+        do {
             distBegin += cursor.getDouble(0);
             if(distBegin > cutDistBegin) {
                 cutColIdBegin = cursor.getInt(1);
                 break;
             }
-            cursor.moveToNext();
-        }
+
+        } while (cursor.moveToNext());
         cursor.moveToLast();
-        while(true) {
+        do {
             distEnd += cursor.getDouble(0);
             if(distEnd > cutDistEnd) {
                 cutColIdEnd = cursor.getInt(1);
                 break;
             }
-            cursor.moveToPrevious();
-        }
+        } while (cursor.moveToPrevious());
         cursor.close();
-        int delRowsBegin = db.delete(TABLENAME, COLUMN_ID + " < " + cutColIdBegin, null);
-        int delRowsEnd = db.delete(TABLENAME, COLUMN_ID + " > " + cutColIdEnd, null);
-        Log.i(TAG, "Random cut: begin=" + delRowsBegin);
-        Log.i(TAG, "Random cut: end=" + delRowsEnd);
+        if ((cutColIdBegin != -1) && (cutColIdEnd != -1)) {
+            int delRowsBegin = db.delete(TABLENAME, COLUMN_ID + " < " + cutColIdBegin, null);
+            int delRowsEnd = db.delete(TABLENAME, COLUMN_ID + " > " + cutColIdEnd, null);
+            Log.i(TAG, "Random cut: begin=" + delRowsBegin);
+            Log.i(TAG, "Random cut: end=" + delRowsEnd);
+        }
         numRows = getNumRows();
         Log.i(TAG, "DB has " + numRows + " rows");
 

